return in finally swallows exceptions

In https://github.com/sympy/sympy/blob/eda48cf0c17afc05e67dcf641d8d216892d8e466/sympy/testing/runtests.py#L282 there is a `return` statement in a `finally` block, which would swallow any in-flight exception.

This means that if an unhandled exception is raised from the `try` body, or any exception is raised from an `except:` clause, it will not propagate on as expected.

See also https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions.

It seems that B012 is the rule that would find this:
```
sympy/testing/runtests.py:282:9: B012 `return` inside `finally` blocks cause exceptions to be silenced
    |
280 |         else:
281 |             os.environ["PYTHONHASHSEED"] = hash_seed
282 |         return p.returncode
    |         ^^^^^^^^^^^^^^^^^^^ B012
    |

Found 1 error.
```

--- a/sympy/testing/runtests.py
+++ b/sympy/testing/runtests.py
@@ -256,8 +256,8 @@ def run_in_subprocess_with_hash_randomization(
                      (module, function, function, repr(function_args),
                       repr(function_kwargs)))
 
+    p = subprocess.Popen([command, "-R", "-c", commandstring], cwd=cwd)
     try:
-        p = subprocess.Popen([command, "-R", "-c", commandstring], cwd=cwd)
         p.communicate()
     except KeyboardInterrupt:
         p.wait()
@@ -268,7 +268,7 @@ def run_in_subprocess_with_hash_randomization(
             del os.environ["PYTHONHASHSEED"]
         else:
             os.environ["PYTHONHASHSEED"] = hash_seed
-        return p.returncode
+    return p.returncode
 
 
 def run_all_tests(test_args=(), test_kwargs=None,

